python/pyspark/sql/readwriter.py

```diff
@@ -367,16 +370,19 @@ def format(self, source):
     @since(1.5)
     def option(self, key, value):
         """Adds an output option for the underlying data source.
+
+        >>> csvpath = os.path.join(tempfile.mkdtemp(), 'data')
+        >>> df.write.option('quote', None).format('csv').save(csvpath)
```

Comment:
`.option("quote", null)` should not work for CSV data source for both reading and writing. I just tested them to double-check. How come this passes tests?

This should throw an exception as below:

```bash
java.lang.NullPointerException was thrown.
java.lang.NullPointerException
	at org.apache.spark.sql.execution.datasources.csv.CSVOptions.getChar(CSVOptions.scala:32)
	at org.apache.spark.sql.execution.datasources.csv.CSVOptions.<init>(CSVOptions.scala:68)
	at org.apache.spark.sql.execution.datasources.csv.DefaultSource.prepareWrite(DefaultSource.scala:86)
...
```